From 54b4231941be4517abb027b8c7ea1abddff7ce99 Mon Sep 17 00:00:00 2001
From: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
Date: Thu, 2 May 2019 15:50:30 +0530
Subject: [PATCH 7/7] usb: xhci: dbc: Reading serial number from kernel CMDLINE

Adding support to read platform serial number from the kernel cmdline
argument for DbC RAW driver.

Tracked-On: OAM-69212
Signed-off-by: Prabhat Chand Pandey <prabhat.chand.pandey@intel.com>
---
 drivers/usb/host/xhci-dbgraw.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-dbgraw.c b/drivers/usb/host/xhci-dbgraw.c
index caa050698eb8..b1445ccf6007 100644
--- a/drivers/usb/host/xhci-dbgraw.c
+++ b/drivers/usb/host/xhci-dbgraw.c
@@ -22,6 +22,10 @@
 
 static DEFINE_IDR(dbc_minors);
 
+static char *serialno = DBC_STR_SERIAL;
+module_param(serialno, charp, S_IRUGO | S_IWUSR);
+MODULE_PARM_DESC(serialno, "Use Platform Serial No. as DbC device serial No.");
+
 struct dbc_dev {
 	struct mutex dev_excl;
 	struct mutex read_excl;
@@ -333,7 +337,6 @@ static struct dbc_function raw_func = {
 	.string = {
 		.manufacturer = DBC_STR_MANUFACTURER,
 		.product = DBC_STR_PRODUCT,
-		.serial = DBC_STR_SERIAL,
 	},
 	.protocol = DBC_PROTOCOL,
 	.vid = DBC_VENDOR_ID,
@@ -347,6 +350,7 @@ static struct dbc_function raw_func = {
 
 static int __init xhci_dbc_raw_init(void)
 {
+	strcpy(&raw_func.string.serial, serialno);
 	return xhci_dbc_register_function(&raw_func);
 }
 
-- 
2.21.0

